### PR TITLE
optional reusable connection

### PR DESCRIPTION
**Pull request by theromis**
My postgres out of connections after continuous PGVector usage, and the reason because it constantly creates new connections, so adding a reusable pre established connection seems like solves an issue

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ def __init__(
         pre_delete_collection: bool = False,
         logger: Optional[logging.Logger] = None,
         relevance_score_fn: Optional[Callable[[float], float]] = None,
+        connection: sqlalchemy.engine.Connection = None,
```
Comment (eyurtsev) [inline libs/langchain/langchain/vectorstores/pgvector.py:117]:
```suggestion
        *,
        connection: sqlalchemy.engine.Connection = None,
```

**Review by eyurtsev — APPROVED** [review]

**Comment by eyurtsev** [comment]
@theromis looks good  i added a `*` -- better later than never

**Comment by eyurtsev** [comment]
@theromis would you be able to resolve the mypy error?

**Review by theromis — COMMENTED** [review]
looks good to me

**Comment by theromis** [comment]
@eyurtsev hopefully this will fixes mypy

**Comment by eyurtsev** [comment]
should be ready to merge when tests pass

**Comment by theromis** [comment]
@eyurtsev all checks passed, I think it ready to merge

**Pull request merged**